app/controllers/charges_controller.rb

```diff
@@ -1,6 +1,6 @@
 class ChargesController < ApplicationController
   def new
-    order = Order.find_by(id: session[:checkout_order_id])
+    order = current_user.orders.pending.last
```

Comment:
Nice change. I'd like to see this moved into a `#current_order` or similar method on the user class (okay for a later pr)